src/TSTransformer/nodes/class/transformDecorators.ts

```diff
@@ -142,7 +142,7 @@ function transformParameterDecorators(
 	const result = luau.list.make<luau.Statement>();
 
 	const memberName = member.name;
-	if (!memberName || ts.isPrivateIdentifier(memberName)) return result;
+	assert(memberName);
```

Comment:
This is also wrong - see the earlier review comment about `constructor` name handling. If `memberName` is undefined, it's from a constructor.

Reply:
Do we want to throw if `memberName` is undefined then? I tried testing this change in a project and it seems to crash, so I think I might be misunderstanding what you mean.

Since TypeScript doesn't allow constructors to be decorated, I figured that `transformParameterDecorators` shouldn't be called for constructors, but it seems to be called for all constructors instead.

Reply:
> doesn't allow constructors to be decorated

Not the constructors themselves, but parameters of them are fine!
[Playground example](https://www.typescriptlang.org/play?experimentalDecorators=true&target=99&ts=5.2.2#code/FAYwNghgzlAECCsDexZtiA9gOygFwCcBXEPTAgCgAEIMAuWfAgS2wHMAaWABweyIC2AIwCmBAJTJYAX2CzgAMyLZSzHLAgVMQgFYNtOkaS7YIAkQyas2sAD6xlAExELWIx12Z9BoicmByQA)

Reply:
Thanks, didn't realise this was the case.

I've changed the implementation so that constructors are now handled. Do we still want to assert that the member isn't a private identifier, or is this not necessary?

Reply:
It seems unnecessary with earlier catches and safe onward execution, but at the same time harmless. No preference from my side.